src/rdb.c

```diff
@@ -2811,56 +2796,77 @@ void rdbLoadProgressCallback(rio *r, const void *buf, size_t len) {
  *
  * The lib_ctx argument is also optional. If NULL is given, only verify rdb
  * structure with out performing the actual functions loading. */
-int rdbFunctionLoad(rio *rdb, int ver, functionsLibCtx* lib_ctx, int rdbflags, sds *err) {
+int rdbFunctionLoad(rio *rdb, int ver, functionsLibCtx* lib_ctx, int type, int rdbflags, sds *err) {
     UNUSED(ver);
-    sds name = NULL;
-    sds engine_name = NULL;
-    sds desc = NULL;
-    sds blob = NULL;
-    uint64_t has_desc;
     sds error = NULL;
+    sds final_payload = NULL;
     int res = C_ERR;
-    if (!(name = rdbGenericLoadStringObject(rdb, RDB_LOAD_SDS, NULL))) {
-        error = sdsnew("Failed loading library name");
-        goto error;
-    }
+    if (type == RDB_OPCODE_FUNCTION) {
+        /* RDB that was generated on versions 7.0 rc1 and 7.0 rc2 has another
+         * an old format that contains the library name, engine and description.
+         * To support this format we must read those values. */
```

Comment:
I'm not sure we have to support that. Upgrade from earlier 7.0 RC would be broken in other ways as well (master/slave replication, AOF, etc.).

Reply:
Will be happy to remove that :), @oranagra WDYT?

Reply:
We did have this situation in redis 4.0.0, the GA had different RDB format than the RCs.
i think we can afford to keep this few lines of code and make it possible for people to upgrade (although they'll need to update their application code which will get broken by this change).
then i think we can delete this code in 7.0.3.

i won't object to deleting it.. just think it's a middle ground of allowing some upgrade path, at least for the data files.